pandora/transport.py

```diff
@@ -46,7 +46,11 @@ def function(*args, **kwargs):
                     retries_left -= 1
                     return func(*args, **kwargs)
 
-                except exceptions:
+                except exceptions as e:
+                    # Don't retry for PandoraExceptions - unlikely that result
+                    # will change for same set of input parameters.
+                    if isinstance(e, PandoraException):
```

Comment:
I think it might make sense to introduce an exception hierarchy so that you can catch only non-retryable exceptions. I'll submit a patch for you to review.